backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/cgen/CBridgeGen.kt

```diff
@@ -1045,6 +1049,34 @@ private class StructValuePassing(private val kotlinClass: IrClass, override val
 
 }
 
+/**
+ * To avoid memory copying, we get global struct by passing it through the bridge by pointer
```

Comment:
> To avoid memory copying

IIRC, the original motivation was to simplify access to struct's fields (`CValue` has problems with that).